### ISSUE TITLE
test_SeqIO_write.py skips test failures

### ISSUE THREAD

**Issue by mdehoon** [biopython/biopython#2823]
See the `check_write_fails` method in `test_SeqIO_write.py`:

```
    def check_write_fails(self, records, format, err_type, err_msg=""):
        if format in SeqIO._BinaryFormats:
            handle = BytesIO()
        else:
            handle = StringIO()
        if err_msg:
            try:
                with warnings.catch_warnings():
                    warnings.simplefilter("ignore", BiopythonWarning)
                    SeqIO.write(records, handle, format)
            except err_type as err:
                self.assertEqual(str(err), err_msg)
        else:
            self.assertRaises(err_type, SeqIO.write, records, handle, format)
        handle.close()
```

If there is an `err_msg`, then we never check if the error is actually raised.

**Comment by mdehoon** [comment]
Apparently two test failures are being missed. If we just use
```
            self.assertRaises(err_type, SeqIO.write, records, handle, format)
```
regardless of `err_msg`, then these two test failures show up:
```
======================================================================
FAIL: test_maf_alignment_with_repeated_record (__main__.WriterTests)
maf for alignment with repeated record
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_SeqIO_write.py", line 173, in <lambda>
    err_msg)
  File "test_SeqIO_write.py", line 141, in check_write_fails
    self.assertRaises(err_type, SeqIO.write, records, handle, format)
AssertionError: ValueError not raised by write

======================================================================
FAIL: test_mauve_3_DNA_seq_alignment_with_CR/LF_in_name/descr (__main__.WriterTests)
mauve for 3 DNA seq alignment with CR/LF in name/descr
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_SeqIO_write.py", line 173, in <lambda>
    err_msg)
  File "test_SeqIO_write.py", line 141, in check_write_fails
    self.assertRaises(err_type, SeqIO.write, records, handle, format)
AssertionError: ValueError not raised by write

----------------------------------------------------------------------
```

**Comment by mdehoon** [comment]
The first test should have raised an `ValueError`. However, since commit ba065bd on April 18, 2015 we allow multiple sequences with the same ID. So this test can be removed.

The second test seems to be a simple mistake. The expected error message is "Sequences have different lengths, or repeated identifier" but the sequences all have the same length, and the identifiers are all different.

**Comment by mdehoon** [comment]
Well actually the second test turns out to be more complicated. `SeqIO.write` does not raise a `ValueError` but writes a mangled file; `SeqIO.read` does raise a `ValueError` but the test is designed for `SeqIO.write`, not `SeqIO.read`.

**Comment by peterjc** [comment]
I don't immediately see what is wrong in the test code snippet you quoted, but glad you're digging into this - evidently something wrong here.

**Comment by mdehoon** [comment]
@peterjc  `check_write_fails` does not notice if `SeqIO.write` fails to raise an error.

**Comment by peterjc** [comment]
Closed by #2824?
